### DISCUSSION TITLE
recompile with -fPIE eror when using build-static.sh

### DISCUSSION THREAD

**Discussion by ghost** [php/frankenphp#1205] [Q&A]
Hi! I'm using build-static.sh to build a single executable file on a Linux machine. i get the following issue.

/frankenphp/dist/static-php-cli/source/php-src/ext/standard/basic_functions.c:2256:(.text+0x561a): warning: Using 'getservbyport' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking

/usr/bin/ld: /usr/lib/gcc/x86_64-redhat-linux/14/libgomp.a(target.o): relocation R_X86_64_32 against `.rodata' can not be used when making a PIE object; recompile with -fPIE
/usr/bin/ld: failed to set dynamic section sizes: bad value
collect2: error: ld returned 1 exit status

**Comment by withinboredom** [comment]
What version of php are you building against? It looks like you are building against a php version built with glibc instead of musl?
